src/persist/src/postgres.rs

```diff
@@ -492,16 +495,160 @@ impl Consensus for PostgresConsensus {
     }
 }
 
+const BLOB_SCHEMA: &str = "
+CREATE TABLE IF NOT EXISTS blob (
+    prefix STRING(1024) NOT NULL,
+    key STRING(1024) NOT NULL,
+    data BYTEA NOT NULL,
+    PRIMARY KEY(prefix, key)
+);
+";
+
+/// Configuration to connect to a Postgres backed implementation of [Blob].
+#[derive(Clone, Debug)]
+pub struct PostgresBlobConfig {
+    url: String,
+    prefix: String,
+}
+
+impl PostgresBlobConfig {
+    /// Returns a new [PostgresBlobConfig] for use in production.
+    pub fn new(url: String, prefix: String) -> Result<Self, Error> {
+        Ok(PostgresBlobConfig { url, prefix })
+    }
+
+    /// Returns a new [PostgresBlobConfig] for use in unit tests.
+    ///
+    /// By default, persist tests that use external storage (like Postgres) are
+    /// no-ops so that `cargo test` works on new environments without any
+    /// configuration. To activate the tests for [PostgresBlob] set the
+    /// `MZ_PERSIST_EXTERNAL_STORAGE_TEST_POSTGRES_URL` environment variable
+    /// with a valid connection url [1].
+    ///
+    /// [1]: https://docs.rs/tokio-postgres/latest/tokio_postgres/config/struct.Config.html#url
+    pub fn new_for_test() -> Result<Option<Self>, Error> {
+        let url = match std::env::var(PostgresConsensusConfig::EXTERNAL_TESTS_POSTGRES_URL) {
+            Ok(url) => url,
+            Err(_) => {
+                if mz_ore::env::is_var_truthy("CI") {
+                    panic!("CI is supposed to run this test but something has gone wrong!");
+                }
+                return Ok(None);
+            }
+        };
+
+        // Give each test a unique prefix so they don't conflict. WIP delete
+        // data somehow.
+        let prefix = uuid::Uuid::new_v4().to_string();
+        let config = PostgresBlobConfig { url, prefix };
+        Ok(Some(config))
+    }
+}
+
+/// Implementation of [Blob] over a Postgres database.
+pub struct PostgresBlob {
+    prefix: String,
+    client: tokio_postgres::Client,
+    _handle: JoinHandle<()>,
+}
+
+impl std::fmt::Debug for PostgresBlob {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PostgresBlob").finish_non_exhaustive()
+    }
+}
+
+impl PostgresBlob {
+    /// Open a Postgres [Blob] instance with `config`, for the collection
+    /// named `shard`.
+    pub async fn open(config: PostgresBlobConfig) -> Result<Self, ExternalError> {
+        let pg_config: tokio_postgres::Config = config.url.parse()?;
+        let tls = make_tls(&pg_config)?;
+        let (mut client, conn) = tokio_postgres::connect(&config.url, tls)
```

Comment:
This would certainly need to be able to reconnect, so should either the actual connection and checking in the Blob impl functions, or else use deadpool or something.

Reply:
yeah, I kinda felt like maybe it should share the pool with PostgresConsensus or something? didn't want to deal thinking through the tradeoffs for a demo, so just did the obvious thing